src/lib/apiTypesV2/Entity.cpp

```diff
@@ -61,6 +62,128 @@ Entity::~Entity()
 }
 
 
+/* ****************************************************************************
+*
+* Entity::filterAttributes -
+*
+* Filter attributes vector in order to get the effective attribute vector to
+* render.
+*
+* dateCreatedOption and dateModifiedOption are due to deprecated ways of requesting
+* date in response. If used, the date is added at the end.
+*
+* FIXME P7: some comments regarding this function
+*
+* - Code should be integrated with the one on ContextElement::filterAttributes(). However,
+*   this is probable a sub-task of the more general task of changing ContextElement to use
+*   Entity class internally
+* - The function is pretty long and complex. It should be refactored to simplify it
+* - Related with this simplification maybe dateCreated and dateModified should be pre-included
+*   in the attributes vector, so all three (dateCreated, dateModified and dateExpires) could be
+*   processed in a similar way.
+* - This function shouldn't be called from toJson() code. The mongoBackend should deal with
+*   attributeVector preparation. Thus, either we leave the function heare and mongoBackend calls
+*   Entity::filterAttributes() or the logic is moved to mongoBackend (to be decided when we face
+*   this FIXME).
+*/
+void Entity::filterAttributes
+(
+  const std::vector<std::string>&  attrsFilter,
+  bool                             dateCreatedOption,
+  bool                             dateModifiedOption
+)
+{
+  bool dateCreatedAdded  = false;
+  bool dateModifiedAdded = false;
+
+  if (attrsFilter.size () != 0)
+  {
+    if (std::find(attrsFilter.begin(), attrsFilter.end(), ALL_ATTRS) != attrsFilter.end())
+    {
+      // Not filtering, just adding dateCreated and dateModified if needed
+      if ((creDate != 0) && (std::find(attrsFilter.begin(), attrsFilter.end(), DATE_CREATED) != attrsFilter.end()))
+      {
+        ContextAttribute* caP = new ContextAttribute(DATE_CREATED, DATE_TYPE, creDate);
+        attributeVector.push_back(caP);
+        dateCreatedAdded = true;
+      }
+      if ((modDate != 0) &&  (std::find(attrsFilter.begin(), attrsFilter.end(), DATE_MODIFIED) != attrsFilter.end()))
+      {
+        ContextAttribute* caP = new ContextAttribute(DATE_MODIFIED, DATE_TYPE, modDate);
+        attributeVector.push_back(caP);
+        dateModifiedAdded = true;
+      }
+    }
+    else
+    {
+      // Reorder attributes in the same order they are in attrsFilter, excluding the ones
+      // note there (i.e. filtering them out) and giving special treatment to creation
+      // and modification dates
```

Comment:
excluding the ones note there ...

Difficult to understand this phrase.
Is it supposed to be "excluding the ones *not * there"?
Doesn't make too much sense. Can't exclude something that is already not present ...

Perhaps:
"except those that aren't present, of course ..."

Reply:
Rewritten in 295296b